Float.Core/UX/ICoordinator.cs

```diff
@@ -23,5 +24,12 @@ public interface ICoordinator
         /// </summary>
         /// <param name="navigationContext">The navigation context for this coordinator.</param>
         void Start(INavigationContext navigationContext);
+
+        /// <summary>
+        /// Requests that a coordinator will finish.
+        /// </summary>
+        /// <param name="eventArgs">The event args.</param>
+        /// <returns>A task, with a boolean indicating whether this did finish.</returns>
```

Comment:
@EBusch I wonder if a boolean is the right return value here. I'm not sure what the caller is supposed to do with `false` -- because it's not necessarily that the call was unsuccessful, it may have simple been deferred. Thinking out loud: I wonder if either...
a. This async method should only return once Finish is successful and throw an exception if it cannot be finished.
b. This method should _not_ be async and return an enum that indicates whether finishing happened immediately, deferred, or unknown/failed.